adapters/yandex/yandex.go

```diff
@@ -0,0 +1,311 @@
+	var ext adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &ext); err != nil {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("imp %s: unable to unmarshal ext", imp.ID),
+		}
+	}
+
+	if err := json.Unmarshal(ext.Bidder, yandexExt); err != nil {
+		return &errortypes.BadInput{
+			Message: fmt.Sprintf("imp %s: unable to unmarshal ext.bidder", imp.ID),
+		}
+	}
+
+	return nil
+}
+
+func modifyImp(imp *openrtb2.Imp) error {
+	if imp.Banner != nil {
+		banner, err := modifyBanner(*imp.Banner)
+		if banner != nil {
+			imp.Banner = banner
+		}
+		return err
+	}
+
+	if imp.Native != nil {
+		return nil
+	}
+
+	return &errortypes.BadInput{
+		Message: fmt.Sprintf("Unsupported format. Yandex only supports banner and native types. Ignoring imp id #%s", imp.ID),
+	}
+}
+
+func modifyBanner(banner openrtb2.Banner) (*openrtb2.Banner, error) {
+	height := banner.H
+	width := banner.W
+	format := banner.Format
+
+	if width == nil || height == nil || *width == 0 || *height == 0 {
+		if len(format) == 0 {
+			return nil, &errortypes.BadInput{
+				Message: "Invalid size provided for Banner",
+			}
+		}
+
+		firstFormat := format[0]
+		banner.H = &firstFormat.H
+		banner.W = &firstFormat.W
+	}
+
+	return &banner, nil
+}
+
+// "Un-templates" the endpoint by replacing macroses and adding the required query parameters
+func (a *adapter) resolveUrl(yandexExt openrtb_ext.ExtImpYandex, referer string, currency string) (string, error) {
+	publisherID := strconv.Itoa(int(yandexExt.PageID))
+	impID := strconv.Itoa(int(yandexExt.ImpID))
+	params := macros.EndpointTemplateParams{PublisherID: publisherID}
+
+	endpointStr, err := macros.ResolveMacros(a.endpoint, params)
+	if err != nil {
+		return "", err
+	}
+
+	parsedUrl, err := url.Parse(endpointStr)
+	if err != nil {
+		return "", err
+	}
+
+	addNonEmptyQueryParams(parsedUrl, map[string]string{
+		refererQueryKey:  referer,
+		currencyQueryKey: currency,
+		impIdQueryKey:    impID,
+	})
+
+	return parsedUrl.String(), nil
+}
+
+func addNonEmptyQueryParams(url *url.URL, queryMap map[string]string) {
+	query := url.Query()
+	for key, value := range queryMap {
+		if len(value) > 0 {
+			query.Add(key, value)
+		}
+	}
+
+	url.RawQuery = query.Encode()
+}
+
+func getReferer(request *openrtb2.BidRequest) string {
+	if request.Site == nil {
+		return ""
+	}
+
+	return request.Site.Domain
+}
+
+func getCurrency(request *openrtb2.BidRequest) string {
+	if len(request.Cur) == 0 {
+		return ""
+	}
+
+	return request.Cur[0]
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, _ *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d.", responseData.StatusCode),
+		}}
+	}
```

Comment:
https://github.com/prebid/prebid-server/blob/ec729e6e5d50a1306d76332fda52108b9d313c1d/adapters/response.go#L10-L28

@chernodub  could make use of above utils

Reply:
Thanks, exactly what was needed here!